DEVELOPER.rst

```diff
@@ -1,71 +1,148 @@
 
-This is from the python tutorial_.
+After you have finished all code for the pull request, make sure to add the
+corresponding changelog files in ``./changelog/``. This
+project uses ``towncrier`` to generate the ``CHANGELOG.rst`` found in the
+project root.
 
-We can generate a source distribution package
+Using ``towncrier``
+-------------------
 
-.. code::
+Every feature that has been added or every bug that has been fixed, requires a
+separate file in the ``./changelog/`` directory (called fragments). Filenames
+should start with the issue or pull request number and either end with
+``.bugfix`` or ``.feature``. For some more possibilities, refer to the
+`towncrier README`_.
 
-   python setup.py sdist
+For example, if you have opened up pull request #20 and fixed a bug in there,
+the corresponding file should be named ``20.bugfix``. This file should contain a
+one-line summary of what was done, as well as a reference to the issue or pull
+request::
 
-We will also build wheels. Since we are pure python a universal wheel is possible
+    Improved plotting functionality. Plots are even more fancy now. (PR #20)
 
-.. code::
+Make sure to add every notable change into a separate fragment and commit all of
+them into your pull request.
 
-   pip install --upgrade wheel
-   python setup.py bdist_wheel --universal
+==================
+Creating a Release
+==================
 
-Check that the source tarball in *dist* contains all important files!
+To create a new release for PyPI and conda, you need to follow a few steps.
 
-Publish on Github
+1. Update version
 -----------------
 
-mark tag on github an release
+Bump the version info in ``mdbenchmark/__init__.py``.
 
-Publish to PyPi
----------------
-upload with twine
+This project uses the `semantic versioning scheme`_.
+
+2. Update ``CHANGELOG``
+-----------------------
 
-.. code::
+To update the ``CHANGELOG``, you first need to install the python package ``towncrier``::
 
-   twine upload dist/*
+    $ pip install towncrier
 
-Publish to Conda-Forge
+Next run ``towncrier``, which should grab all changelog files from
+``./changelog/`` and add their contents to ``CHANGELOG.rst``::
+
+    $ towncrier
+
+You can also dry-run ``towncrier``, thus not making any changes::
+
+    $ towncrier --draft
+
+After ``towncrier`` has finished, the folder ``./changelog/`` should only
+contain the file ``.invisible``. Also the file ``CHANGELOG.rst`` should have
+been updated.
+
+3. Generate dist files
 ----------------------
 
-After the pypi upload update the conda-forge recipe
+Next we need to actually build the project. The following steps were taken from
+the official `python packaging guide`_.
+
+First make sure that your ``wheel`` package is up-to-date::
+
+    $ pip install --upgrade wheel
+
+Next we can generate a source distribution package and universal wheel::
+
+   $ python setup.py sdist bdist_wheel --universal
+
+Check that the source tarball in ``./dist/`` contains all important files!
```

Comment:
This would be the actual source code. The Readme and license file at the very minimum.

Reply:
Oh no also the CHANGELOG.  And check that the version on the `__init__.py` and changelog match. I made that mistake with the last release.